Views/TweetCell.m

```diff
@@ -21,4 +24,107 @@ - (void)setSelected:(BOOL)selected animated:(BOOL)animated {
+        self.tweet.favoriteCount -= 1;
+        // TODO: Send a POST request to the POST favorites/create endpoint
+         [[APIManager shared] unfavorite:self.tweet completion:^(Tweet *tweet, NSError *error) {
+             if(error){
+                  NSLog(@"Error unfavoriting tweet: %@", error.localizedDescription);
+             }
+             else{
+                 NSLog(@"Successfully unfavoriting the following Tweet: %@", tweet.text);
+             }
+         }];
+    }
+    else
+    {
+        self.tweet.favorited = YES;
+        self.tweet.favoriteCount += 1;
+        // TODO: Send a POST request to the POST favorites/create endpoint
+         [[APIManager shared] favorite:self.tweet completion:^(Tweet *tweet, NSError *error) {
+             if(error){
+                  NSLog(@"Error favoriting tweet: %@", error.localizedDescription);
+             }
+             else{
+                 NSLog(@"Successfully favoriting the following Tweet: %@", tweet.text);
+             }
+         }];
+    }
+    [self refreshData];
+}
+
+- (IBAction)didTapRetweet:(id)sender {
+    NSLog(@"tapped retweet");
+    if(self.tweet.retweeted)
+    {
+        self.tweet.retweeted = NO;
+        self.tweet.retweetCount -= 1;
+        // TODO: Send a POST request to the POST favorites/create endpoint
+         [[APIManager shared] unretweet:self.tweet completion:^(Tweet *tweet, NSError *error) {
+             if(error){
+                  NSLog(@"Error unretweeting tweet: %@", error.localizedDescription);
+             }
+             else{
+                 NSLog(@"Successfully unretweeting the following Tweet: %@", tweet.text);
+             }
+         }];
+    }
+    else
+    {
+        self.tweet.retweeted = YES;
+        self.tweet.retweetCount += 1;
+        // TODO: Send a POST request to the POST favorites/create endpoint
+         [[APIManager shared] retweet:self.tweet completion:^(Tweet *tweet, NSError *error) {
+             if(error){
+                  NSLog(@"Error retweet tweet: %@", error.localizedDescription);
+             }
+             else{
+                 NSLog(@"Successfully retweet the following Tweet: %@", tweet.text);
+             }
+         }];
+    }
+    [self refreshData];
+}
+
+- (void)refreshData {
+    //MARK: setting everything inside the tweet cell
+    //profile image
+    NSString *URLString = self.tweet.user.profilePicture;
+    NSURL *url = [NSURL URLWithString:URLString];
+    NSData *urlData = [NSData dataWithContentsOfURL:url];
+    self.pfpView.image = nil;           //clear previous image if it takes too long to load
+    [self.pfpView setImageWithURL:url];
+    
+    //username
+    self.usernameLabel.text = self.tweet.user.name;
+    
+    //username handle
+    self.usernameHandleLabel.text = [@"@" stringByAppendingString: self.tweet.user.screenName];
+    
+    //timestamp/date published
+    self.timestampLabel.text = self.tweet.createdAtString;
+    
+    //the actual tweet
```

Comment:
I would delete all of these comments. Also, the tweet is the tweet. This is the text of the tweet.